src/jit/morph.cpp

```diff
@@ -11571,7 +11571,18 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             break;
 
         case GT_DIV:
-
+            /* Replace "val / dcon" with "val * (1.0 / dcon)" if dcon is a power of two */
+            if ((typ == TYP_FLOAT || typ == TYP_DOUBLE) && op2->IsCnsFltOrDbl())
+            {
+                double op2dVal = op2->gtDblCon.gtDconVal;
+                int64_t op2iVal = static_cast<int64_t>(op2dVal);
+                if (op2dVal == op2iVal && isPow2(op2iVal))
+                {
+                    oper = GT_MUL;
+                    tree->ChangeOper(oper);
+                    op2->gtDblCon.gtDconVal = 1.0 / op2dVal;
```

Comment:
This is a bit problematic if denormals are treated as zero for whatever reason (someone PInvoked code that changed the FP environment). Perhaps it would be better that together with a proper power of 2 check to also reject exponents 1023 (for double) and 127 (for float).